File: 20-valid-parentheses/valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack=[]
        for ch in s:
            if ch in '([{':
                stack.append(ch)
                continue
            elif ch==')' and (stack==[] or stack[-1]!='('):
                return False
            elif ch==']' and (stack==[] or stack[-1]!='['):
                return False
            elif ch=='}' and (stack==[] or stack[-1]!='{'):
                return False
            stack.pop()
        if stack==[]: return True
        else: return False

File: 20-valid-parentheses/valid-parentheses.java
class Solution {
    public boolean isValid(String s) {

        Stack<Character> stack=new Stack<>();

        for(char ch: s.toCharArray()){

            if(ch=='(' || ch=='{' || ch=='['){
                stack.push(ch);
            }
            else if(ch==')' && (stack.isEmpty()==true || stack.pop()!='(')){
                return false;
            }
             else if(ch=='}' && (stack.isEmpty()==true || stack.pop()!='{')){
                return false;
            }
             else if(ch==']' && (stack.isEmpty()==true || stack.pop()!='[')){
                return false;
            }
        }

        return stack.isEmpty();
       
    }
}